packages/react-popover/src/components/PopoverTrigger/usePopoverTrigger.ts

```diff
@@ -82,17 +67,17 @@ export const usePopoverTrigger = (
     child.props?.onMouseLeave?.(e);
   });
 
-  const child = React.Children.only(state.children);
-  state.children = React.cloneElement(child, {
-    'aria-haspopup': 'true',
-    onClick,
-    onMouseEnter,
-    onKeyDown,
-    onMouseLeave,
-    onContextMenu,
-    ref: useMergedRefs(((child as unknown) as React.ReactElement & React.RefAttributes<unknown>).ref, triggerRef),
-    ...triggerAttributes,
-  });
-
-  return state;
+  const child = React.Children.only(props.children);
+  return {
+    children: React.cloneElement(child, {
+      'aria-haspopup': 'true',
+      onClick,
+      onMouseEnter,
+      onKeyDown,
+      onMouseLeave,
+      onContextMenu,
+      ref: useMergedRefs(((child as unknown) as React.ReactElement & React.RefAttributes<unknown>).ref, triggerRef),
+      ...triggerAttributes,
```

Comment:
It's not related to this PR, but order of merging there is wrong, it's the same problem as #18875 fixed.

---

The proper order of props:

- DOM attributes from Tabster / ARIA
- user's `props` to make them overridable
- merged overrides (callback handlers, `ref`)

⬇⬇⬇

```ts
React.cloneElement(child, {
  ...triggerAttributes,
  "aria-haspopup": "true",
  ...child.props,
  onClick,
  // ...
  ref
});
```

_Why it's needed?_

Otherwise user will not have any way to override our attributes:

```tsx
<PopoverTrigger>
  {/* It's **wrong**, but there might be a usecase when this is needed */}
  <div aria-haspopup="false" />
</PopoverTrigger>
```

With current implementation we will get `aria-haspopup="true"` always.